tests/test_mysql_binlog.py

```diff
@@ -0,0 +1,705 @@
+            our_bit_1              BIT(1),
+            our_datetime           DATETIME,
+            our_timestamp          TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+            our_date               DATE,
+            our_time               TIME,
+            our_boolean            BOOLEAN
+)
+""".format(self.database_name(), self.table_name())
+
+            cur.execute(create_table_sql)
+
+            for record in [rec_1, rec_2]:
+                self.insert_record(cur, record)
+
+
+    def test_run(self):
+        conn_id = connections.ensure_connection(self)
+
+        # run in check mode
+        check_job_name = runner.run_check_mode(self, conn_id)
+
+        # verify check exit codes
+        exit_status = menagerie.get_exit_status(conn_id, check_job_name)
+        menagerie.verify_check_exit_status(self, exit_status, check_job_name)
+
+        expected_check_streams = { self.tap_stream_id() }
+        expected_sync_streams = { self.table_name() }
+        expected_pks = {
+            self.table_name(): { 'id' }
+        }
+
+        # verify the tap discovered the right streams
+        found_catalogs = [catalog for catalog
+                          in menagerie.get_catalogs(conn_id)
+                          if catalog['tap_stream_id'] in expected_check_streams]
+
+        self.assertGreaterEqual(len(found_catalogs),
+                                1,
+                                msg="unable to locate schemas for connection {}".format(conn_id))
+
+        found_catalog_names = set(map(lambda c: c['tap_stream_id'], found_catalogs))
+        diff = expected_check_streams.symmetric_difference(found_catalog_names)
+        self.assertEqual(len(diff), 0, msg="discovered schemas do not match: {}".format(diff))
+
+        # verify that persisted streams have the correct properties
+        test_catalog = found_catalogs[0]
+
+        self.assertEqual(self.table_name(), test_catalog['stream_name'])
+
+        print("discovered streams are correct")
+
+        additional_md = [{ "breadcrumb" : [], "metadata" : {'replication-method' : 'LOG_BASED'}}]
+        selected_metadata = connections.select_catalog_and_fields_via_metadata(conn_id,
+                                                                               test_catalog,
+                                                                               menagerie.get_annotated_schema(conn_id, test_catalog['stream_id']),
+                                                                               additional_md)
+
+        # clear state
+        menagerie.set_state(conn_id, {})
+
+        # run initial full table sync
+        sync_job_name = runner.run_sync_mode(self, conn_id)
+
+        exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
+        menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
+
+        # verify the persisted schema was correct
+        records_by_stream = runner.get_records_from_target_output()
+
+        self.maxDiff = None
```

Comment:
Can you move this into the setup method? And comment what it does for those unfamiliar?

Reply:
I don't quite want to make any changes to the test, maybe in another cleanup PR since this was a move. We need to get the tests passing first though.

Reply:
I didn't realize these tests were copied over. I agree you should not make changes as part of the same PR which is mostly copying and pasting changes from another repo.